core/include/seqan/align/evaluate_alignment.h

```diff
@@ -252,6 +252,241 @@ TScoreVal computeAlignmentStats(Align<TGaps, TAlignSpec> const & align,
     return computeAlignmentStats(stats, align, scoringScheme);
 }
 
+// ----------------------------------------------------------------------------
+// Function alignmentIdentityString()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn Align#alignmentIdentityString()
+ * @headerfile <seqan/align.h>
+ * @brief Returns a CharString of the similarity of positions in an alignment
+ *
+ * @signature CharString alignmentIdentityString();
+ *
+ * @param align The @link Align @endlink object to score for similarity.
+ * @param match The character to use for similarity at match positions
+ * @param mismatch The character to use for similarity at mismatch positions
+ * @param gap The character to use for similarity at gap positions
+ *
+ * @signature CharString A string of characters representing alignment position similarity
+ *
+ * @section Examples
+ *
+ * @include demos/align/compute_match_string.cpp
+ *
+ * The output is as follows:
+ *
+ * @include demos/align/compute_match_string.cpp.stdout
+ */
+
+
+template <typename TGaps>
+String<char> alignmentIdentityString(TGaps & queryRow,
+                                     TGaps & referenceRow,
+                                     char const matchChar,
+                                     char const mismatchChar,
+                                     char const gapChar)
+{
+    // Define short-hand names for various types to be used
+    typedef typename Iterator<TGaps, Standard>::Type TGapsIter;
+    typedef typename Value<typename Source<TGaps>::Type>::Type TAlphabet;
+
+    // Define and size the char string we will return
+    String<char> matchString;
+    unsigned alignLength = std::min(length(queryRow), length(referenceRow));
+    resize(matchString, alignLength, Exact());
+
+    // Get iterators.
+    TGapsIter it0 = begin(queryRow);
+    TGapsIter itEnd0 = end(queryRow);
+    TGapsIter it1 = begin(referenceRow);
+    TGapsIter itEnd1 = end(referenceRow);
+
+    // Counter for the position in the similarity string
+    unsigned pos = 0;
+
+    for (; it0 != itEnd0 && it1 != itEnd1; ++it0, ++it1, ++pos)
+    {
```

Comment:
Remove everything from above and only keep the matchString definition. 
Also this function then gets as third parameter the functor not the chars to write the string.
These are then stored within the functor.
Then call transform() as defined in the comment above and return matchString.